src/vuex/modules/session.js

```diff
@@ -105,6 +104,12 @@ export default () =>
           await commit(`setUserAddresses`, JSON.parse(addresses))
         }
       },
+      async checkForPersistedAddressRole({ commit }) {
```

Comment:
why?

Reply:
remove and always query?

Reply:
state does not persist

Reply:
you refresh and boom! restrictions are blown away

Reply:
if you query for the role on any sign in it will always be available?

Reply:
if we persist it in localStorage yes. Or I don't know if I'm following you

Reply:
I think we should never persist the role, should be fetched in every sign-in as the user can i.e import the address one day in Lunie, change the controller in Polkadot JS another day, return to Lunie and boom!

Reply:
don't store in local storage. run the query every time. we also don't know if the status of the account changes. one might add or change the controller account later.

Reply:
Then run it every time in TmBalance. Still needs to be persisted though. What happens if the user just refreshes the page in PageValidator for instance? I hope I succeed pointing at the issue here

Reply:
Maybe we can query in TmSessionSignIn and store in session?

Reply:
> mean the function signIn in session.js is triggered when the user refresh the page, right?

yes